bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/handler/RoomHandler.java

```diff
@@ -112,41 +133,82 @@ public void refresh(@Nullable QueryResponseDTO domain) {
         }
     }
 
-    private void setOverride(final QuantityType<?> command) {
-        String roomId = getThing().getProperties().get(PROPERTY_ROOM_ID);
-        String locationId = getThing().getProperties().get(PROPERTY_LOCATION_ID);
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(WarmupActions.class);
+    }
 
-        QuantityType<?> temp = command.toUnit(SIUnits.CELSIUS);
+    private void setOverride(final QuantityType<?> command) {
+        setOverride(command, new QuantityType<>(config.getOverrideDuration(), Units.MINUTE));
+    }
 
-        if (temp != null) {
-            final int value = temp.multiply(BigDecimal.TEN).intValue();
+    public void setOverride(final QuantityType<?> temperature, final QuantityType<?> duration) {
+        setOverride(formatTemperature(temperature), duration.toUnit(Units.MINUTE).intValue());
+    }
 
+    private void setOverride(final int temperature, final int duration) {
+        if (duration > 1440 || duration <= 0) {
+            logger.info("Set Override failed: duration must be between 0 and 1440 minutes");
+        }
+        if (temperature > 600 || temperature < 50) {
+            logger.info("Set Override failed: temperature must be between 0.5 and 60 degrees C");
+        } else {
             try {
-                final MyWarmupAccountHandler bridgeHandler = getBridgeHandler();
-                if (bridgeHandler != null && config != null) {
-                    final int overrideDuration = config.getOverrideDuration();
-                    if (overrideDuration > 0 && locationId != null && roomId != null) {
-                        bridgeHandler.getApi().setOverride(locationId, roomId, value, overrideDuration);
-                        refreshFromServer();
-                    }
-                }
+                RoomCallout rc = getCallout();
+                rc.api.setOverride(rc.locationId, rc.roomId, temperature, duration);
+                refreshFromServer();
             } catch (MyWarmupApiException e) {
-                logger.debug("Set Override failed: {}", e.getMessage());
+                logger.info("Set Override failed: {}", e.getMessage());
```

Comment:
Should be logged as a warn

Reply:
I set all the logs in this class to `warn` as they're all likely misconfigurations.